speculos.py

```diff
@@ -63,18 +97,20 @@ def run_qemu(s1, s2, app_path, libraries=[], seed=DEFAULT_SEED, debug=False, tra
     if trace_syscalls:
         args += [ '-t' ]
 
-    if rampage is not None and pagesize is not None:
-        args += [ '-r' , rampage ] + [ '-s' , pagesize ]
-
     if model is not None:
         args += ['-m', model]
 
     args += [ '-k', str(sdk_version) ]
 
     for lib in [ f'main:{app_path}' ] + libraries:
         name, lib_path = lib.split(':')
-        load_offset, load_size, stack, stack_size = get_elf_infos(lib_path)
-        args.append(f'{name}:{lib_path}:{load_offset}:{load_size}:{stack}:{stack_size}')
+        load_offset, load_size, stack, stack_size, ram_addr, ram_size = get_elf_infos(lib_path)
+        # Since binaries loaded as libs could also declare extra RAM page(s), collect them all
+        if (ram_addr, ram_size) != (0, 0):
+            extra_ram.append({'addr':ram_addr, 'size':ram_size})
+        args.append(f'{name}:{lib_path}:{hex(load_offset)}:{hex(load_size)}:{hex(stack)}:{hex(stack_size)}')
+
```

Comment:
I guess it's clearer to add the condition `if model == 'blue'` here.

Reply:
Added `model == blue` to the if clause, but did not remove the rest of the clause since it doesn't make sense to consider a zero-byte page starting at address 0.

Reply:
Sorry, I probably commented the wrong line. I would move the `if model == 'blue':` clause outside of `sanitize_ram_args()`:

```
if model == 'blue':
    args.extend(sanitize_ram_args(model, rampage, pagesize, extra_ram))
```

Reply:
done.

Reply:
The condition `if model == 'blue':` in `sanitize_ram_args()` isn't required anymore and can be removed. Thanks!

Reply:
done